addons/account_edi/models/account_move.py

```diff
@@ -135,6 +135,66 @@ def _compute_edi_show_abandon_cancel_button(self):
     # Export Electronic Document
     ####################################################
 
+    def _prepare_edi_vals_to_export(self):
+        ''' The purpose of this helper is to prepare values in order to export an invoice through the EDI system.
+        This includes the computation of the tax details for each invoice line that could be very difficult to
+        handle regarding the computation of the base amount.
+        :return: A python dict containing default pre-processed values.
+        '''
+        self.ensure_one()
+
+        res = {
+            'record': self,
+            'invoice_line_vals_list': [],
+        }
+
+        # Invoice lines details.
+        tax_detail_per_tax = {}
+        added_base_amount_keys = set()
+        for index, line in enumerate(self.invoice_line_ids.filtered(lambda line: not line.display_type), start=1):
+            line_vals = {
+                'index': index,
+                **line._prepare_edi_vals_to_export(),
+            }
+            res['invoice_line_vals_list'].append(line_vals)
+
+            # Tax details.
+            for tax_vals in line_vals['tax_detail_vals_list']:
+                tax_detail_per_tax.setdefault(tax_vals['tax'], {
+                    'tax': tax_vals['tax'],
+                    'src_tax': tax_vals['src_tax'],
+                    'tax_base_amount_currency': 0.0,
+                    'tax_amount_currency': 0.0,
+                    'tag_ids': set(),
+                })
+                vals = tax_detail_per_tax[tax_vals['tax']]
+
+                # Avoid adding multiple times the same base (e.g. with multiple taxes on the same line).
+                base_amount_key = (line.id, tax_vals['tax']['id'])
+                if base_amount_key not in added_base_amount_keys:
+                    vals['tax_base_amount_currency'] += tax_vals['tax_base_amount_currency']
+                    added_base_amount_keys.add(base_amount_key)
+
+                vals['tax_amount_currency'] += tax_vals['tax_amount_currency']
+                for tag in tax_vals['tags']:
```

Comment:
is this why we needed the browse?

Reply:
In MX, we need to display some values depending the tag's name.
This helper is providing the tax detail per line and the tax detail for the whole invoice